docs/tools/prism-eslint-hook.js

```diff
@@ -0,0 +1,170 @@
+"use strict";
+
+const Prism = require("prismjs");
+const { Linter } = require("../../lib/api");
+const astUtils = require("../../lib/shared/ast-utils");
+
+/** @typedef {import("../../lib/shared/types").ParserOptions} ParserOptions */
+
+/**
+ * A set of content that needs to be marked.
+ * @type {Map<string, ParserOptions>}
+ */
+const contentsMustBeMarked = new Map();
+
+/**
+ * Add content that needs to be marked.
+ * @param {string} content Source code content that marks eslint errors.
+ * @param {ParserOptions} options The options used for validation.
+ * @returns {void}
+ */
+function addContentMustBeMarked(content, options) {
+    contentsMustBeMarked.set(content, options);
+}
+
+/**
+ * Register a hook for `Prism` to mark errors in ESLint.
+ * @returns {void}
+ */
+function installPrismESLintMarkerHook() {
+    const linter = new Linter({ configType: "flat" });
+
+    Prism.hooks.add("after-tokenize", env => {
+
+        const parserOptions = contentsMustBeMarked.get(env.code);
+
+        if (!parserOptions) {
+
+            // Ignore unmarked content.
+            return;
+        }
+        contentsMustBeMarked.delete(env.code);
+
+        const code = env.code;
+        const tokens = env.tokens;
+
+        /** Copied from SourceCode constructor */
+        const lineStartIndices = [0];
+        const lineEndingPattern = astUtils.createGlobalLinebreakMatcher();
+        let match;
+
+        while ((match = lineEndingPattern.exec(code))) {
+            lineStartIndices.push(match.index + match[0].length);
+        }
+
+        /**
+         * Converts a (line, column) pair into a range index.
+         * @param {Object} loc A line/column location
+         * @param {number} loc.line The line number of the location (1-indexed)
+         * @param {number} loc.column The column number of the location (1-indexed)
+         * @returns {number} The range index of the location in the file.
+         * Copied from SourceCode#getIndexFromLoc
+         */
+        function getIndexFromLoc(loc) {
+            const lineStartIndex = lineStartIndices[loc.line - 1];
+            const positionIndex = lineStartIndex + loc.column - 1;
+
+            return positionIndex;
+        }
+
+        /*
+         * Run lint to extract the error range.
+         */
+        const messages = linter.verify(
+            code,
```

Comment:
In some examples of the `no-multiple-empty-lines` rule we use the character `⏎` to indicate the presence of a line break, e.g.:

https://github.com/eslint/eslint/blob/5a8efd5b7ad13eb320a1f468d1d4ab3c8ab99214/docs/src/rules/no-multiple-empty-lines.md?plain=1#L61-L70

This character is only useful as a visual guide, it should not be parsed. In fact, it is removed from the code when the rule example is validated or when the Playground link is generated:

https://github.com/eslint/eslint/blob/5a8efd5b7ad13eb320a1f468d1d4ab3c8ab99214/docs/tools/markdown-it-rule-example.js#L75-L78

To get correct results, I think we should apply the same kind of transformation to the `code` that is being linted here.

The fact that the column number could be off by one after this transformation should not be an issue at all, because the `no-multiple-empty-lines` rule only ever reports on empty lines.

Reply:
Thank you for checking this PR!
I fixed an issue related to `no-multiple-empty-lines`.
https://deploy-preview-18041--docs-eslint.netlify.app/rules/no-multiple-empty-lines